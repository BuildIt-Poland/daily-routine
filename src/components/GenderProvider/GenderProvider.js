import React, { useState } from 'react';

import GenderContext from '../../context/GenderContext';
import { MALE } from '../../constants/genders';

const GenderProvider = ({ children }) => {
  const onGenderChange = ({ target }) =>
    setGender(prevState => ({
      ...prevState,
      gender: target.value
    }));

  const genderState = {
    gender: MALE,
    onGenderChange
  };

  const [gender, setGender] = useState(genderState);

  return <GenderContext.Provider value={gender}>{children}</GenderContext.Provider>;
};

export default GenderProvider;
