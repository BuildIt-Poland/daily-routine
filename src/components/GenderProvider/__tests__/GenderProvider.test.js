import React from 'react';
import { create } from 'react-test-renderer';

import GenderProvider from '../GenderProvider';

describe('COMPONENT - GenderProvider', () => {
  it('renders correctly', () => {
    const component = create(<GenderProvider />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
