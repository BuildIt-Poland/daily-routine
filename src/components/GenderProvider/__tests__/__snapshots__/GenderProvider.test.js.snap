// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - GenderProvider renders correctly 1`] = `null`;
